encodings/fastlanes/src/bitpacking/compute/take.rs

```diff
@@ -210,4 +215,25 @@ mod test {
                 );
             });
     }
+
+    #[test]
+    #[cfg_attr(miri, ignore)]
+    fn take_signed_with_patches() {
+        let start = BitPackedArray::encode(
+            &PrimitiveArray::from(vec![1i32, 2i32, 3i32, 4i32]).into_array(),
+            1,
+        )
+        .unwrap();
+
+        let taken_primitive = super::take_primitive::<u32, u64>(
```

Comment:
We usually test blackbox, but had to go whitebox here since the take impl has a configurable UNPACK_THRESHOLD and I didn't want us to stop testing this codepath in case we changed the constant.

This test was able to repro the clickbench error before the fix above to reinterpret_cast before patching